core/corehttp/gateway_handler.go

```diff
@@ -162,7 +116,7 @@ func (i *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		errmsg = errmsg + "bad request for " + r.URL.Path
 	}
 	w.Write([]byte(errmsg))
-	log.Debug(errmsg)
+	log.Error(errmsg) // TODO(cryptix): Why are we ignoring handler errors?
```

Comment:
we should never ignore errors. any time an error is ignored anywhere in our code, its a bug.